Sming/appinit/user_main.cpp

```diff
@@ -77,26 +78,38 @@ extern "C" uint32 ICACHE_FLASH_ATTR  __attribute__((weak)) user_rf_cal_sector_se
 
 extern "C" void ICACHE_FLASH_ATTR  __attribute__((weak)) user_pre_init(void)
 {
-	enum flash_size_map size_map = system_get_flash_size_map();
-	const uint32 rf_cal_addr = user_rf_cal_sector_set() * 0x1000;
-	const uint32 phy_data_addr = rf_cal_addr +  0x1000;
-	const uint32 system_param_addr = phy_data_addr + 0x1000;
+	const uint32_t MAX_PROGRAM_SECTORS = 1024; // 1MB addressable
 
 	// WARNING: Sming supports SDK 3.0 with rBoot enabled apps ONLY!
-#define SYSTEM_PARTITION_RBOOT_CONFIG SYSTEM_PARTITION_CUSTOMER_BEGIN
+	const partition_type_t SYSTEM_PARTITION_RBOOT_CONFIG = static_cast<partition_type_t>(SYSTEM_PARTITION_CUSTOMER_BEGIN + 0);
+	const partition_type_t SYSTEM_PARTITION_PROGRAM = static_cast<partition_type_t>(SYSTEM_PARTITION_CUSTOMER_BEGIN + 1);
 
+	// Partitions offsets and sizes must be in sector multiples, so work in sectors
+	#define PARTITION_ITEM(_type, _start, _length) \
+		{_type, (_start) * SPI_FLASH_SEC_SIZE, (_length) * SPI_FLASH_SEC_SIZE}
+
+	// Partitions in position order
+	uint32_t rfCalSector = user_rf_cal_sector_set();
 	static const partition_item_t partitions[] = {
-		{SYSTEM_PARTITION_BOOTLOADER, 0x0, 0x1000},
-		{SYSTEM_PARTITION_RBOOT_CONFIG, 0x1000, 0x1000},
-		{SYSTEM_PARTITION_RF_CAL, rf_cal_addr, 0x1000},
-		{SYSTEM_PARTITION_PHY_DATA, phy_data_addr, 0x1000},
-		{SYSTEM_PARTITION_SYSTEM_PARAMETER, system_param_addr, 0x3000},
-		{SYSTEM_PARTITION_CUSTOMER_BEGIN, 0x2000, 0xfdff0}, // (1M - 0x2010)
+			PARTITION_ITEM(SYSTEM_PARTITION_BOOTLOADER,			0,					1),
+			PARTITION_ITEM(SYSTEM_PARTITION_RBOOT_CONFIG,		1,					1),
+			PARTITION_ITEM(SYSTEM_PARTITION_PROGRAM,			2,					MAX_PROGRAM_SECTORS - 2),
```

Comment:
There is something fishy here. Now the sample rBoot app refuses to load
```
Booting rom 0.
ota1 not set
ota2 not set
p 101 and p 4 covered: addr:0x2000 + len:0x3fe000 > 0x3fb000 !!!
system_partition_table_regist: failed
size_map = 4
partition[0]: 1, 0x00000000, 0x00001000
partition[1]: 100, 0x00001000, 0x00001000
partition[2]: 101, 0x00002000, 0x003fe000 !!!?  That is wrong. A rom can span up to 1MB. 
partition[3]: 4, 0x003fb000, 0x00001000
partition[4]: 5, 0x003fc000, 0x00001000
partition[5]: 6, 0x003fd000, 0x00003000
```

The size of partition 2, start 0x00002000, cannot be bigger than 1M-0x2000.  With the this PR the size is `0x003fe000` >  1MB.

Reply:
I did an amend commit last night... may have confused things

Reply:
Ammend commits are fine. Make sure to git push -f.

Reply:
OK, cool. Not sure why you're seeing the old version though?

Reply:
> Not sure why you're seeing the old version though?

Probably because the change is still  on your machine only. 

```
cd %SMING_HOME%
git checkout fix/system_partition_table_regist
git status -uno
# if there are any differences that are still not commited you will see them 
# either in the staging area or in the working files.
git commit -a --amend 
git push -f 
```

Reply:
No, it's definitely in the PR - I checked from an anonymous web browser.

Reply:
Ah. I see what I've done. (Confusion between 0x1000 and 1024...)